lsp-metals.el

```diff
@@ -183,6 +183,15 @@ Should be ignored if there is no open doctor window."
                         (c (ignore (lsp-warn "Unknown metals client command: %s" c))))))
     (apply command (append (list workspace) (ht-get params "arguments") nil))))
 
+(defvar lsp-metals--current-buffer nil
```

Comment:
Maybe better to store it in workspace metadata?

Reply:
Is it working fine with this version of the code?

Reply:
Yes, but it will send unnecessary notifications when switching between different metals workspaces.

Reply:
Before sending a message you may check if the buffer is managed by metals instance. As a side note. E. g.
```
(with-current-buffer "HelloSpec.scala"
  (lsp--workspace-buffers (lsp-find-workspace 'metals (buffer-file-name)) ))
```

Reply:
I add local hooks only to metals buffers, so no need to check it.

Reply:
So IIUC 

> it will send unnecessary notifications when switching between different metals workspaces.

is not a problem?

Reply:
Yes, not a problem at all :)